### PR TITLE
feat: kyber dmm

### PR DESCRIPTION
**Pull request by nduchak**
Closes BACK-99

### CODE REVIEW

File: src/dex/kyberdmm.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    destToken: string,
+    srcAmount: string,
+    destAmount: string,
+    data: KyberDmmData,
+    side: SwapSide,
+  ): Promise<SimpleExchangeParam> {
+    const swapFunctionParams: KyberDmmParam = [
+      srcAmount,
+      destAmount,
+      [data.pools[0].address],
```
Comment (shresthagrawal) [inline src/dex/kyberdmm.ts:85]: Shouldn't this be?
```suggestion
      data.pools.map(p => p.address),
```

File: src/dex/kyberdmm.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    side: SwapSide,
+  ): AdapterExchangeParam {
+    const payload = this.abiCoder.encodeParameter(
+      {
+        ParentStruct: {
+          poolPath: 'address[]',
+          path: 'address[]',
+        },
+      },
+      { poolPath: [data.pools[0].address], path: data.path },
```
Comment (shresthagrawal) [inline src/dex/kyberdmm.ts:65]: similar to the comment below
```suggestion
      { poolPath: data.pools.map(p => p.address), path: data.path },
```

File: src/dex/kyberdmm.ts [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+  destToken: string,
+  pools: string[],
+  path: string[],
+  to: string,
+  deadline: string,
+];
+
+enum KyberDMMFunctions {
+  swapExactTokensForTokens = 'swapExactTokensForTokens',
+  swapExactTokensForETH = 'swapExactTokensForETH',
```
Comment (shresthagrawal) [inline src/dex/kyberdmm.ts:30]: We don't use this I think

File: src/dex/kyberdmm.ts (hunk 1 above)
Comment (nduchak) [reply src/dex/kyberdmm.ts:85]: hmm...I thought that we always have one pool, but anyway this should work

**Review by shresthagrawal — APPROVED** [review]

**Pull request merged**